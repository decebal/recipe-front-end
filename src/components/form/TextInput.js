import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import classnames from 'classnames';
import Label from './Label';
import InputFeedback from './InputFeedback';

const StyledInput = styled.input`
  padding: .5rem;
  font-size: 16px;
  width: 100%;
  display: block;
  border-radius: 4px;
  border: 1px solid #ccc;

  &:focus {
    border-color: #007eff;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 0 3px rgba(0, 126, 255, 0.1);
    outline: none;
  }
`;

const PaddedInputGroup = styled.div`
  margin-bottom: 1rem;
`;

const TextInput = ({
  type, id, label, error, value, onChange, className, ...props
}) => {
  const classes = classnames(
    {
      'animated shake error': !!error,
    },
    className,
  );
  return (
    <PaddedInputGroup className={classes}>
      <Label htmlFor={id} error={error}>
        {label}
      </Label>
      <StyledInput
        id={id}
        type={type}
        value={value}
        onChange={onChange}
        {...props}
      />
      <InputFeedback error={error} />
    </PaddedInputGroup>
  );
};

TextInput.propTypes = {
  error: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

TextInput.defaultProps = {
  error: null,
  className: null,
  value: '',
};


export default TextInput;
