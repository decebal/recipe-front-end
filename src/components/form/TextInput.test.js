import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';


test('TextInput renders correctly', () => {
  const tree = renderer
    .create(<TextInput
      id="invalid-uuid-4"
      label="textarea"
      value="Text"
      type="text"
      onChange={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
