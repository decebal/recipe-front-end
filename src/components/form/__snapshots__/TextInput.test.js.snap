// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<div
  className="sc-bxivhb gCIOfD"
>
  <label
    className="sc-bdVaJa bNfbPK"
    htmlFor="invalid-uuid-4"
  >
    textarea
  </label>
  <input
    className="sc-htpNat bPsheZ"
    id="invalid-uuid-4"
    onChange={[Function]}
    type="text"
    value="Text"
  />
</div>
`;
